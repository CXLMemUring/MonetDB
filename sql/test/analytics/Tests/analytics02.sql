start transaction;

create table analytics (aa int, bb int, cc bigint);
insert into analytics values (15, 3, 15), (3, 1, 3), (2, 1, 2), (5, 3, 5), (NULL, 2, NULL), (3, 2, 3), (4, 1, 4), (6, 3, 6), (8, 2, 8), (NULL, 4, NULL);

select cast (sum(aa) over (rows between 5 preceding and 0 following) as bigint) from analytics;
select cast (sum(aa) over (rows between 5 preceding and 2 following) as bigint) from analytics;
select cast (sum(aa) over (partition by bb order by bb rows between 5 preceding and 0 following) as bigint) from analytics;
select cast (sum(aa) over (partition by bb order by bb rows between 5 preceding and 2 following) as bigint) from analytics;

create table debugme (aa real, bb int);
insert into debugme values (15, 3), (3, 1), (2, 1), (5, 3), (NULL, 2), (3, 2), (4, 1), (6, 3), (8, 2), (NULL, 4);

select cast (sum(aa) over (rows between 2 preceding and 0 following) as bigint) from debugme;
select cast (sum(aa) over (rows between 2 preceding and 2 following) as bigint) from debugme;
select cast (sum(aa) over (partition by bb order by bb rows between 2 preceding and 0 following) as bigint) from debugme;
select cast (sum(aa) over (partition by bb order by bb rows between 2 preceding and 2 following) as bigint) from debugme;

rollback;
