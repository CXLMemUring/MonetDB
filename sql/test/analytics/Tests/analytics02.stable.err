stderr of test 'analytics02` in directory 'sql/test/analytics` itself:


# 12:55:22 >  
# 12:55:22 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=38610" "--set" "mapi_usock=/var/tmp/mtest-8804/.s.monetdb.38610" "--set" "monet_prompt=" "--forcemito" "--dbpath=/home/ferreira/repositories/MonetDB-analytics/BUILD/var/MonetDB/mTests_sql_test_analytics" "--set" "embedded_c=true"
# 12:55:22 >  

# builtin opt 	gdk_dbpath = /home/ferreira/repositories/MonetDB-analytics/BUILD/var/monetdb5/dbfarm/demo
# builtin opt 	gdk_debug = 0
# builtin opt 	gdk_vmtrim = no
# builtin opt 	monet_prompt = >
# builtin opt 	monet_daemon = no
# builtin opt 	mapi_port = 50000
# builtin opt 	mapi_open = false
# builtin opt 	mapi_autosense = false
# builtin opt 	sql_optimizer = default_pipe
# builtin opt 	sql_debug = 0
# cmdline opt 	gdk_nr_threads = 0
# cmdline opt 	mapi_open = true
# cmdline opt 	mapi_port = 38610
# cmdline opt 	mapi_usock = /var/tmp/mtest-8804/.s.monetdb.38610
# cmdline opt 	monet_prompt = 
# cmdline opt 	gdk_dbpath = /home/ferreira/repositories/MonetDB-analytics/BUILD/var/MonetDB/mTests_sql_test_analytics
# cmdline opt 	embedded_c = true
# cmdline opt 	gdk_debug = 553648138

# 12:55:22 >  
# 12:55:22 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-8804" "--port=38610"
# 12:55:22 >  


# 12:55:22 >  
# 12:55:22 >  "Done."
# 12:55:22 >  

