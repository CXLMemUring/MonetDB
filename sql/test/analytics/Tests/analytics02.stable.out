stdout of test 'analytics02` in directory 'sql/test/analytics` itself:


# 12:55:22 >  
# 12:55:22 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=38610" "--set" "mapi_usock=/var/tmp/mtest-8804/.s.monetdb.38610" "--set" "monet_prompt=" "--forcemito" "--dbpath=/home/ferreira/repositories/MonetDB-analytics/BUILD/var/MonetDB/mTests_sql_test_analytics" "--set" "embedded_c=true"
# 12:55:22 >  

# MonetDB 5 server v11.32.0
# This is an unreleased version
# Serving database 'mTests_sql_test_analytics', using 8 threads
# Compiled for x86_64-pc-linux-gnu/64bit with 128bit integers
# Found 15.492 GiB available main-memory.
# Copyright (c) 1993 - July 2008 CWI.
# Copyright (c) August 2008 - 2018 MonetDB B.V., all rights reserved
# Visit https://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://wired-142.cwi.nl:38610/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-8804/.s.monetdb.38610
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded

Ready.

# 12:55:22 >  
# 12:55:22 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-8804" "--port=38610"
# 12:55:22 >  

#start transaction;
#create table analytics (aa int, bb int, cc bigint);
#insert into analytics values (15, 3, 15), (3, 1, 3), (2, 1, 2), (5, 3, 5), (NULL, 2, NULL), (3, 2, 3), (4, 1, 4), (6, 3, 6), (8, 2, 8), (NULL, 4, NULL);
[ 10	]
#select cast (sum(aa) over (rows between 5 preceding and 0 following) as bigint) from analytics;
% sys.L5 # table_name
% L5 # name
% bigint # type
% 2 # length
[ 15	]
[ 18	]
[ 20	]
[ 25	]
[ 25	]
[ 28	]
[ 17	]
[ 20	]
[ 26	]
[ 21	]
#select cast (sum(aa) over (rows between 5 preceding and 2 following) as bigint) from analytics;
% sys.L5 # table_name
% L5 # name
% bigint # type
% 2 # length
[ 20	]
[ 25	]
[ 25	]
[ 28	]
[ 32	]
[ 38	]
[ 31	]
[ 28	]
[ 26	]
[ 21	]
#select cast (sum(aa) over (partition by bb order by bb rows between 5 preceding and 0 following) as bigint) from analytics;
% sys.L5 # table_name
% L5 # name
% bigint # type
% 2 # length
[ 3	]
[ 5	]
[ 9	]
[ 3	]
[ 3	]
[ 11	]
[ 15	]
[ 20	]
[ 26	]
[ NULL	]
#select cast (sum(aa) over (partition by bb order by bb rows between 5 preceding and 2 following) as bigint) from analytics;
% sys.L5 # table_name
% L5 # name
% bigint # type
% 2 # length
[ 9	]
[ 9	]
[ 9	]
[ 11	]
[ 11	]
[ 11	]
[ 26	]
[ 26	]
[ 26	]
[ NULL	]
#create table debugme (aa real, bb int);
#insert into debugme values (15, 3), (3, 1), (2, 1), (5, 3), (NULL, 2), (3, 2), (4, 1), (6, 3), (8, 2), (NULL, 4);
[ 10	]
#select cast (sum(aa) over (rows between 2 preceding and 0 following) as bigint) from debugme;
% sys.L5 # table_name
% L5 # name
% bigint # type
% 2 # length
[ 15	]
[ 18	]
[ 20	]
[ 10	]
[ 7	]
[ 8	]
[ 7	]
[ 13	]
[ 18	]
[ 14	]
#select cast (sum(aa) over (rows between 2 preceding and 2 following) as bigint) from debugme;
% sys.L5 # table_name
% L5 # name
% bigint # type
% 2 # length
[ 20	]
[ 25	]
[ 25	]
[ 13	]
[ 14	]
[ 18	]
[ 21	]
[ 21	]
[ 18	]
[ 14	]
#select cast (sum(aa) over (partition by bb order by bb rows between 2 preceding and 0 following) as bigint) from debugme;
% sys.L5 # table_name
% L5 # name
% bigint # type
% 2 # length
[ 3	]
[ 5	]
[ 9	]
[ 3	]
[ 3	]
[ 11	]
[ 15	]
[ 20	]
[ 26	]
[ NULL	]
#select cast (sum(aa) over (partition by bb order by bb rows between 2 preceding and 2 following) as bigint) from debugme;
% sys.L5 # table_name
% L5 # name
% bigint # type
% 2 # length
[ 9	]
[ 9	]
[ 9	]
[ 11	]
[ 11	]
[ 11	]
[ 26	]
[ 26	]
[ 26	]
[ NULL	]
#rollback;

# 12:55:22 >  
# 12:55:22 >  "Done."
# 12:55:22 >  

